### PR TITLE
Most of the steps to get a static Linux build working.

### PR DESCRIPTION
**Pull request by evetion**
In `scripts/compile_deps.sh` there's a commented out section which compiles meshkernel, with a `vim build/CMakeCache.txt`.

In there I manually had to set:
`OpenMP_CXX_FLAGS:STRING=-fopenmp` -> `OpenMP_CXX_FLAGS:STRING=-fopenmp -static`
`OpenMP_C_FLAGS:STRING=-fopenmp` -> `OpenMP_C_FLAGS:STRING=-fopenmp -static`
`OpenMP_gomp_LIBRARY:FILEPATH=/some/path/libgomp.so` -> `OpenMP_gomp_LIBRARY:FILEPATH=/some/path/libgomp.a`

Profiling/coverage information is still in the Linux builds, should be nicely removed on a Release build.

### CODE REVIEW

**Review by ahmad-el-sayed — CHANGES_REQUESTED** [review]

File: scripts/compile_deps.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export CXX=/opt/rh/devtoolset-9/root/usr/bin/g++
+
+# add cmake
+cd /root
+wget https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1-linux-x86_64.sh
+chmod +x cmake-3.23.1-linux-x86_64.sh
+mkdir /opt/cmake
+./cmake-3.23.1-linux-x86_64.sh --skip-license --prefix=/opt/cmake
+
+# add gcc
```
[the remark this answers is not in the document]
Comment (lucacarniato) [reply scripts/compile_deps.sh:16]: I removed the commented block

File: CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +59,9 @@ include(FetchContent)
 if(WIN32)
   # In windows use static libraries
   set(Boost_USE_STATIC_LIBS ON)
+else()
```
[the remark this answers is not in the document]
Comment (lucacarniato) [reply CMakeLists.txt:62]: i will rebase with branch feature/GRIDEDIT-484_remove_boost_system_and_filesystem_dependencies

File: scripts/compile_deps.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#wget https://ftp.gnu.org/gnu/gcc/gcc-9.3.0/gcc-9.3.0.tar.gz --no-check-certificate
+#tar -xzf gcc-9.3.0.tar.gz
+#cd ./gcc-9.3.0
+#./contrib/download_prerequisites
+#./configure CFLAGS="-fPIC" CXXFLAGS="-fPIC" --enable-languages=c,c++ --disable-multilib
+#cd /root/gcc-9.3.0
+#make -j6
+#make install
+#cd ..
+#rm -rf gcc-9.3.0.tar.gz
```
[the remark this answers is not in the document]
Comment (lucacarniato) [reply scripts/compile_deps.sh:26]: this part of the code is removed

File: scripts/compile_deps.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+# add boost
```
[the remark this answers is not in the document]
Comment (lucacarniato) [reply scripts/compile_deps.sh:28]: done

**Comment by lucacarniato** [comment]
Now linux build is working in master

**Pull request closed without merging**